opengrok-web/src/main/java/org/opengrok/web/CookieFilter.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ */
+package org.opengrok.web;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.HttpHeaders;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Enumeration;
+
+/**
+ * Makes sure that all cookies originating from the web application have the Same-site attribute set.
+ */
+public class CookieFilter implements Filter {
+    FilterConfig fc;
+
+    @Override
+    public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
+            throws IOException, ServletException {
+
+        HttpServletResponse response = (HttpServletResponse) res;
+
+        chain.doFilter(req, response);
+
+        // Change the existing cookies to use the attributes and values from the configuration.
+        Collection<String> headers = response.getHeaders(HttpHeaders.SET_COOKIE);
+        boolean firstHeader = true;
+        for (String header : headers) { // there can be multiple Set-Cookie attributes
+            if (firstHeader) {
+                response.setHeader(HttpHeaders.SET_COOKIE, String.format("%s; %s", header, getSuffix(response)));
+                firstHeader = false;
+                continue;
+            }
+            response.addHeader(HttpHeaders.SET_COOKIE, String.format("%s; %s", header, getSuffix(response)));
+        }
+    }
+
+    private String getSuffix(HttpServletResponse response) {
+        StringBuilder sb = new StringBuilder();
+
+        for (Enumeration<String> e = fc.getInitParameterNames(); e.hasMoreElements();) {
```

Comment:
Would this work?

```java
StreamSupport.stream(Spliterators.spliteratorUnknownSize(fc.getInitParameterNames(), Spliterator.NONNULL), false)
        .map(attributeName -> {
            String attributeValue = fc.getInitParameter(attributeName);
            if (!attributeValue.isEmpty()) {
                return attributeName + "=" + attributeValue;
            } else {
                return attributeValue;
            }
        }).collect(Collectors.joining("; "));
```

Reply:
The first argument of `spliteratorUnknownSize` has to be either primitive number type or `Iterator` which is not available in JDK 8. Also, this sort of conversion to stream seems, uh, unnatural.

Reply:
Yes, however, there are helper methods for this in Guava and maybe apache-commons -> if we have on of them on the classpath then we could leverage that. If not, then creating a simple `toStream` util method would make this nicer.

But you are right, https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Enumeration.html#asIterator() is available since JDK9